include/maidsafe/routing/client.h

```diff
@@ -92,65 +98,85 @@ class Client : public std::enable_shared_from_this<Client> {
   asio::io_service& io_service_;
   const Address our_id_;
   const asymm::Keys our_keys_;
-  boost::optional<Address> bootstrap_node_;
   std::atomic<MessageId> message_id_;
+  boost::optional<Address> bootstrap_node_;
   BootstrapHandler bootstrap_handler_;
+  std::vector<PeerNode> connected_peers_;
   LruCache<std::pair<Address, MessageId>, void> filter_;
   Sentinel sentinel_;
 };
 
 template <typename CompletionToken>
-BootstrapReturn<CompletionToken> Client::Bootstrap(CompletionToken token) {
-  auto handler(std::forward<decltype(token)>(token));
-  auto result(handler);
+BootstrapReturn<CompletionToken> Client::Bootstrap(CompletionToken&& token) {
```

Comment:
What is the reason for && here is it purely for perfect forwarding and why do we want that here?